### ISSUE TITLE
tokenization produces corrupted features vectors for fulltext

### ISSUE THREAD

**Issue by aoboturov** [kermitt2/grobid#214]
C++ isspace() would treat \r\t\n as symbols which would be removed in Wapiti's `src/reader.c`:

		uint32_t cnt = 0;
		while (*line != '\0') {
			toks[cnt++] = line;
			while (*line != '\0' && !isspace(*line))
				line++;
			if (*line == '\0')
				break;
			*line++ = '\0';
			while (*line != '\0' && isspace(*line))
				line++;
		}

We have PDF2XML which actually produces such sequences: `´\rÓÑÔÙØØØ`. In return, it would be set up to the text value of org.grobid.core.layout.LayoutToken. And this text would be used to serialize the the org.grobid.core.features.FeaturesVectorFulltext.printVector()

**Comment by aoboturov** [comment]
In particular, the problem is with the GrobidDefaultAnalyzer#tokenize.

**Comment by aoboturov** [comment]
Example document:

[AS_136494979227648@1409554347281_content_1.pdf](https://github.com/kermitt2/grobid/files/1194608/AS_136494979227648.1409554347281_content_1.pdf)

**Comment by kermitt2** [comment]
Just looking at it now ... the problem is that we only consider the "standard" space character when checking if a token is a space (when a token is a space, end-of-line or `\n`, it will be ignored when producing the feature vectors).

Basically it is fixed simply by changing the line 370 in `FullTextParser.java` - `text` is the textual content of the token (so it might be also any delimiters)

```
text = text.replaceAll(" ", "");
```  

by 

```
text = text.replaceAll("\\s+", "");
```

then if the text length is 0, the token is skipped when generating the feature vector. 

Ideally we should check with all the possible unicode equivalent to a space, an end of line, etc.

**Comment by aoboturov** [comment]
fixed in #216 #218.
